lib/pyspark_init.py

```diff
@@ -0,0 +1,249 @@
+        collect_list("genres").alias("genres"),
+        first("category").alias("category"),
+        collect_list("characters").alias("characters"),
+        first("primaryName").alias("primaryName"),
+        first("knownForTitles").alias("knownForTitles"),
+    )
+    return data
+
+
+def add_kaggle_data(spark: SparkSession, data: DataFrame) -> DataFrame:
+    oscars = spark.read.option("header", "true").csv("the_oscar_award.csv")
+    globe = spark.read.option("header", "true").csv("golden_globe_awards.csv")
+    # emmy_awards_category = spark.read.option("header", "true").csv(
+    #     "emmy_awards_categories.csv"
+    # )
+    emmy_awards = spark.read.option("header", "true").csv("the_emmy_awards.csv")
+    # tmdb_credits=spark.read.option("header","true").csv('tmdb_5000_credits.csv')
+    # tmdb_movies=spark.read.option("header","true").csv('tmdb_5000_movies.csv')
+    oscars_selected = oscars.filter(
+        (oscars.category.like("%ACTOR%")) | (oscars.category.like("%ACTRESS%"))
+    )
+    globe_selected = globe.filter(
+        (globe.category.like("%Actor%")) | (globe.category.like("%Actress%"))
+    )
+    actor_categories = [
+        "Outstanding Lead Actor in a Comedy Series",
+        "Outstanding Lead Actor in a Drama Series",
+        "Outstanding Lead Actor in a Limited or Anthology Series or Movie",
+        "Outstanding Lead Actress in a Comedy Series",
+        "Outstanding Lead Actress in a Drama Series",
+        "Outstanding Lead Actress in a Limited or Anthology Series or Movie",
+        "Outstanding Supporting Actor in a Comedy Series",
+        "Outstanding Supporting Actor in a Drama Series",
+        "Outstanding Supporting Actor in a Limited or Anthology Series or Movie",
+        "Outstanding Supporting Actress in a Comedy Series",
+        "Outstanding Supporting Actress in a Drama Series",
+        "Outstanding Supporting Actress in a Limited or Anthology Series or Movie",
+        "Outstanding Character Voice-Over Performance",
+        "Outstanding Guest Actor in a Drama Series",
+        "Outstanding Guest Actor in a Comedy Series",
+        "Outstanding Guest Actress in a Drama Series",
+        "Outstanding Guest Actress in a Comedy Series",
+        "Outstanding Narrator",
+        "Outstanding Actor in a Short Form Comedy or Drama Series",
+        "Outstanding Actress in a Short Form Comedy or Drama Series",
+        "Best Specialty Act – Single or Group",
+        "Outstanding Voice-Over Performance",
+        "Outstanding Sports Personality",
+        "Most Outstanding Live Personality",
+        "Most Outstanding Kinescoped Personality",
+    ]
+    emmy_awards_selected = emmy_awards.filter(
+        emmy_awards.category.isin(actor_categories)
+    )
+    emmy_awards_selected = emmy_awards_selected.withColumn(
+        "staff", explode(split("staff", ", "))
+    )
+    emmy_awards_selected = emmy_awards_selected.groupby("nominee", "staff").agg(
+        first("id").alias("id"),
```

Comment:
potwierdźcie mi, że te first są celowe. Tabele mają po prostu duplikaty tych samych wartości?

Reply:
W tej kolumnie jest zapisane osoba, rola. Czasem widzę, że jest zapisanych w ten sposób kilka osób obok siebie. Możemy brać pod uwagę wszystkie wartości, a role po prostu nie będą się łączyć z niczym. Bo tak to rzeczywiście bierzemy tylko jedną osobę